test/cql-pytest/cassandra_tests/distinct_query_paging_test.py

```diff
@@ -0,0 +1,179 @@
+# This file was translated from the original Java test from the Apache
+# Cassandra source repository, as of commit 2a4cd36475de3eb47207cd88d2d472b876c6816d
+#
+# The original Apache Cassandra license:
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from cassandra_tests.porting import *
+
+from cassandra.query import BatchStatement, BatchType
+from cassandra.protocol import InvalidRequest
+
+# Migrated from cql_tests.py:TestCQL.test_select_distinct()
+def testSelectDistinct(cql, test_keyspace):
+    # Test a regular (CQL3) table.
+    with create_table(cql, test_keyspace, "(pk0 int, pk1 int, ck0 int, val int, PRIMARY KEY ((pk0, pk1), ck0))") as table:
+        for i in range(3):
+            execute(cql, table, "INSERT INTO %s (pk0, pk1, ck0, val) VALUES (?, ?, 0, 0)", i, i)
+            execute(cql, table, "INSERT INTO %s (pk0, pk1, ck0, val) VALUES (?, ?, 1, 1)", i, i)
+
+        assertRows(execute(cql, table, "SELECT DISTINCT pk0, pk1 FROM %s LIMIT 1"),
+                   row(0, 0))
+
+        assertRows(execute(cql, table, "SELECT DISTINCT pk0, pk1 FROM %s LIMIT 3"),
+                   row(0, 0),
+                   row(2, 2),
+                   row(1, 1))
```

Comment:
Can you please help me understand why such order of rows? I'd expect them to be ordered by pk0 ascending values.